stories/actions/update.stories.tsx

```diff
@@ -0,0 +1,52 @@
+import { Meta } from '@storybook/react';
+import { Layout } from '../utils/Layout';
+import {
+  RoiContainer,
+  RoiList,
+  RoiProvider,
+  useCommittedRois,
+  useRoiActions,
+  useRoiState,
+} from '../../src';
+import { Image } from '../utils/Image';
+import { initialRois } from './utils';
+
+export default {
+  title: 'Actions',
+} as Meta;
+
+export function Update() {
+  return (
+    <Layout>
+      <RoiProvider initialRois={initialRois}>
+        <div style={{ display: 'flex', flexDirection: 'column', gap: 5 }}>
+          <Internal />
+
+          <RoiContainer target={<Image />}>
+            <RoiList />
+          </RoiContainer>
+        </div>
+      </RoiProvider>
+    </Layout>
+  );
+}
+
+function Internal() {
+  const { selectedRoi } = useRoiState();
+  const committedRois = useCommittedRois();
+  const { updateRoi } = useRoiActions();
+
+  function onClick(type: 'start' | 'end') {
+    const selected = committedRois.find((r) => r.id === selectedRoi);
+    if (!selected) return;
```

Comment:
This check should not be necessary. If it is it's a bug.

You only need `selectedRoi` and `updateRoi`. If there is no selected roi the `selectedRoi` is undefined. If it's defined it should exist in the state and can be updated.

Reply:
i think, i need selected too (and with a find, he can be undefined).
Because, i need to get the current width of the roi to place on "x" the correct value 1 - roiWidth to make the roi at the end of the image.

If i just hardcode 1 it will be placed out of the image and will teleport to the right place a few second after.
And if i do not calculate with the width, if i change the width of the roi, i can't be sure that it will be at the end.

Reply:
The fact that it "teleports" is a bug.

You can simplify by just moving the roi to the left and to the top and create an issue about this.